specs/_features/maxeb_increase/capella.py

```diff
@@ -1627,6 +1652,43 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
     state.pending_balance_deposits = state.pending_balance_deposits[next_pending_deposit_index:]
 
 
+def apply_pending_consolidation(state: BeaconState, pending_consolidation: PendingConsolidation) -> None:
+    target_validator = state[consolidation.target_index]
+    source_validator = state[consolidation.source_index]
+
+    # Ensure the source and the target exit statuses have not changed
+    if source_validator.exit_epoch != FAR_FUTURE_EPOCH:
+        return
+    if target_validator.exit_epoch != FAR_FUTURE_EPOCH:
+        return
```

Comment:
Looks good to me overall! 
Only thing I am unsure about is whether this part here is safe:

```suggestion
    # Ensure the source and the target exit statuses have not changed
    if source_validator.exit_epoch != FAR_FUTURE_EPOCH:
        return
    if target_validator.exit_epoch != FAR_FUTURE_EPOCH:
        return
```

In principle we want to ensure that a finalized consolidation will be processed the same way on any branch, and this seems to violate that, e.g., on one branch I exit the source validator and on another one I don't, resulting in the consolidation being applied on one but not the other

Reply:
Applied a more complex logic which applies consolidation disregarding of exit statuses of the source and the target with the corresponding consequences in all possible cases